File: Console/BExIS.Web.Shell/Models/ModuleModels.cs
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using Vaiona.Utils.Cfg;

namespace BExIS.Web.Shell.Models
{
    public class ReadModuleModel
    {
        [JsonProperty("id")]
        public string Id { get; set; }

        [JsonProperty("title")]
        public string Title { get; set; }

        [JsonProperty("description")]
        public string Description { get; set; }

        [JsonProperty("settings")]
        public JsonSettings Settings { get; set; }
    }

    public class ModulModel
    {
        [JsonProperty("id")]
        public string Id { get; set; }

        [JsonProperty("title")]
        public string Title { get; set; }

        [JsonProperty("description")]
        public string Description { get; set; }
    }
}

File: Console/BExIS.Web.Shell.Svelte/src/models/moduleModels.ts
import { ReadSettingModel } from './settingModels';

export class ReadModuleModel {
	id: string;
	title: string;
	description: string;
	settings: ReadSettingModel;

	constructor(json: any) {
		this.id = json.id;
		this.title = json.title;
		this.description = json.description;
		this.settings = json.settings;
	}
}
